From aa065fe910290e0fcc426d4a55cfc73bad0b2b6d Mon Sep 17 00:00:00 2001
From: Lubos Kocman <Lubos.Kocman@suse.com>
Date: Tue, 29 Oct 2024 14:28:26 +0100
Subject: [PATCH] openSUSE distro refresh

* Add openSUSE Slowroll
* Updated logos for openSUSE Leap and Tumbleweed
  gh#openSUSE/branding#154 gh#openSUSE/branding#160
* Update neofetch accordingly to gh#openSUSE/neofetch fork
---
 hyfetch/distros/opensuse_leap.py       | 34 ++++-----
 hyfetch/distros/opensuse_slowroll.py   | 27 ++++++++
 hyfetch/distros/opensuse_tumbleweed.py | 35 ++++++----
 neofetch                               | 95 ++++++++++++++++++--------
 4 files changed, 133 insertions(+), 58 deletions(-)
 create mode 100644 hyfetch/distros/opensuse_slowroll.py

diff --git a/hyfetch/distros/opensuse_leap.py b/hyfetch/distros/opensuse_leap.py
index b39b18665da..a784d848eff 100644
--- a/hyfetch/distros/opensuse_leap.py
+++ b/hyfetch/distros/opensuse_leap.py
@@ -3,21 +3,23 @@
 from . import AsciiArt
 
 opensuse_leap = AsciiArt(match=r'''"openSUSE Leap"* | "openSUSE_Leap"*''', color='2 7', ascii=r"""
-${c2}                 `-++:`
-               ./oooooo/-
-            `:oooooooooooo:.
-          -+oooooooooooooooo+-`
-       ./oooooooooooooooooooooo/-
-      :oooooooooooooooooooooooooo:
-    `  `-+oooooooooooooooooooo/-   `
- `:oo/-   .:ooooooooooooooo+:`  `-+oo/.
-`/oooooo:.   -/oooooooooo/.   ./oooooo/.
-  `:+ooooo+-`  `:+oooo+-   `:oooooo+:`
-     .:oooooo/.   .::`   -+oooooo/.
-        -/oooooo:.    ./oooooo+-
-          `:+ooooo+-:+oooooo:`
-             ./oooooooooo/.
-                -/oooo+:`
-                  `:/.
+${c2}            ====
+           ======
+         ==== ====+
+       +====    +====
+     +===+        ====
+    ====            ====
+  +===               +====
+  ====               +====
+   =====            ====
+     +===+        =====
+  ==+  =====    +===+  ===
+  ====   ==== =====  =====
+    ====  =======   ====
+      ====  ===   ====
+       ====+    ====
+         ==== =====
+           ======
+             ==
 """)
     
\ No newline at end of file
diff --git a/hyfetch/distros/opensuse_slowroll.py b/hyfetch/distros/opensuse_slowroll.py
new file mode 100644
index 00000000000..3d140782e37
--- /dev/null
+++ b/hyfetch/distros/opensuse_slowroll.py
@@ -0,0 +1,27 @@
+# This file is automatically generated. Please do not modify.
+
+from . import AsciiArt
+
+opensuse_slowroll = AsciiArt(match=r'''"openSUSE Slowroll"* | "openSUSE_Slowroll"*''', color='2 7', ascii=r"""
+${c2}              _aaaymQQmwaaa,
+          ,wWQQQD????????$QQQQa,.
+       _wQQB?"              ??QQQa,
+     sQQD^                      ?QQ6\
+    yWW'                          4QQg
+  ,QQD          .aaaaaaaa          ^4Q6
+ ,mQP        _wWQW?????YWWQa,        4Qm
+ jQ@        wWW?'        ^4QQc       ^$QL
+,QQ'       jWW'            )QW\       ]QQ
+|QQ       ,QW'              ]QQ       ^QQ|
+|QQ       |QQ               ]QQ        QQ|
+|QQ        4Qg              ]QQ       .QQ|
+'QQ6       '$WQac.         _QQ(       jQQ
+ ]QQw        "?QWQQf      _mQP       ,QQ(
+  4QQga                  wQQP       ,mQ?
+   4QQQga,            saQWP'       jQQf
+    ?QQQQQQwaaaaaaaayWWW?'       _mQ@'
+      ?WQQQP?9VWUV???^        _amQP^
+        "4QQQaa,          ,awQQQ?^
+           "?VQQQQQQQQQQQQQQP?'
+""")
+    
\ No newline at end of file
diff --git a/hyfetch/distros/opensuse_tumbleweed.py b/hyfetch/distros/opensuse_tumbleweed.py
index f8f5a130053..163c564424f 100644
--- a/hyfetch/distros/opensuse_tumbleweed.py
+++ b/hyfetch/distros/opensuse_tumbleweed.py
@@ -3,18 +3,27 @@
 from . import AsciiArt
 
 opensuse_tumbleweed = AsciiArt(match=r'''"openSUSE Tumbleweed"* | "openSUSE_Tumbleweed"*''', color='2 7', ascii=r"""
-${c2}                                     ......
-     .,cdxxxoc,.               .:kKMMMNWMMMNk:.
-    cKMMN0OOOKWMMXo. A        ;0MWk:'      ':OMMk.
-  ;WMK;'       'lKMMNM,     :NMK'             'OMW;
- cMW;             WMMMN   ,XMK'                 oMM.
-.MMc             ''^*~l. xMN:                    KM0
-'MM.                   .NMO                      oMM
-.MM,                 .kMMl                       xMN
- KM0               .kMM0' .dl>~,.               .WMd
- 'XM0.           ,OMMK'    OMMM7'              .XMK
-   *WMO:.    .;xNMMk'       NNNMKl.          .xWMx
-     ^ONMMNXMMMKx;          V  'xNMWKkxllox0NMWk'
-         '''''                    ':dOOXXKOxl'
+${c2}         JJJJJJJJ
+      JJJJJJJJJJJJJJ
+    JJJJJJ   =JJJJJJJ
+   JJJJ      =JJJ JJJJ
+   JJJ       =JJJ   JJJ
+  JJJJ       =JJJ   JJJ
+  JJJJJJJJJJJJJJJ   JJJJ
+   JJJJJJJJJJJJJJ   JJJJ
+   JJJJ             JJJJ
+    JJJJJ=          JJJJ
+      JJJJJJJJJJJJJJJJJJJJJJJJJJJJJ=
+        =JJJJJJJJJJJJJJJJJJJJJJJJJJJJJ
+                    JJJJ         =JJJJJJ
+                    JJJJ            =JJJJ
+                    JJJJ   JJJJJJJJJJJJJJ
+                    JJJJ   JJJJJJJJJJJJJJJ
+                    JJJJ   JJJJ       JJJJ
+                     JJJ   JJJJ       JJJ
+                     JJJJJ JJJJ      JJJJ
+                      =JJJJJJJJ   JJJJJJ
+                        JJJJJJJJJJJJJJ
+                           JJJJJJJ=
 """)
     
\ No newline at end of file
diff --git a/neofetch b/neofetch
index d6e9273cc99..32601b5ce94 100755
--- a/neofetch
+++ b/neofetch
@@ -14437,22 +14437,24 @@ EOF
         "openSUSE Leap"* | "openSUSE_Leap"*)
             set_colors 2 7
             read -rd '' ascii_data <<'EOF'
-${c2}                 `-++:`
-               ./oooooo/-
-            `:oooooooooooo:.
-          -+oooooooooooooooo+-`
-       ./oooooooooooooooooooooo/-
-      :oooooooooooooooooooooooooo:
-    `  `-+oooooooooooooooooooo/-   `
- `:oo/-   .:ooooooooooooooo+:`  `-+oo/.
-`/oooooo:.   -/oooooooooo/.   ./oooooo/.
-  `:+ooooo+-`  `:+oooo+-   `:oooooo+:`
-     .:oooooo/.   .::`   -+oooooo/.
-        -/oooooo:.    ./oooooo+-
-          `:+ooooo+-:+oooooo:`
-             ./oooooooooo/.
-                -/oooo+:`
-                  `:/.
+${c2}            ====
+           ======
+         ==== ====+
+       +====    +====
+     +===+        ====
+    ====            ====
+  +===               +====
+  ====               +====
+   =====            ====
+     +===+        =====
+  ==+  =====    +===+  ===
+  ====   ==== =====  =====
+    ====  =======   ====
+      ====  ===   ====
+       ====+    ====
+         ==== =====
+           ======
+             ==
 EOF
         ;;
 
@@ -14469,22 +14471,57 @@ TTTTTTTTTT
 EOF
         ;;
 
+        "openSUSE Tumbleweed-Slowroll"* | "openSUSE_Slowroll"*)
+            set_colors 2 7
+            read -rd '' ascii_data <<'EOF'
+${c2}              _aaaymQQmwaaa,
+          ,wWQQQD????????$QQQQa,.
+       _wQQB?"              ??QQQa,
+     sQQD^                      ?QQ6\
+    yWW'                          4QQg
+  ,QQD          .aaaaaaaa          ^4Q6
+ ,mQP        _wWQW?????YWWQa,        4Qm
+ jQ@        wWW?'        ^4QQc       ^$QL
+,QQ'       jWW'            )QW\       ]QQ
+|QQ       ,QW'              ]QQ       ^QQ|
+|QQ       |QQ               ]QQ        QQ|
+|QQ        4Qg              ]QQ       .QQ|
+'QQ6       '$WQac.         _QQ(       jQQ
+ ]QQw        "?QWQQf      _mQP       ,QQ(
+  4QQga                  wQQP       ,mQ?
+   4QQQga,            saQWP'       jQQf
+    ?QQQQQQwaaaaaaaayWWW?'       _mQ@'
+      ?WQQQP?9VWUV???^        _amQP^
+        "4QQQaa,          ,awQQQ?^
+           "?VQQQQQQQQQQQQQQP?'
+EOF
+        ;;
+
         "openSUSE Tumbleweed"* | "openSUSE_Tumbleweed"*)
             set_colors 2 7
             read -rd '' ascii_data <<'EOF'
-${c2}                                     ......
-     .,cdxxxoc,.               .:kKMMMNWMMMNk:.
-    cKMMN0OOOKWMMXo. A        ;0MWk:'      ':OMMk.
-  ;WMK;'       'lKMMNM,     :NMK'             'OMW;
- cMW;             WMMMN   ,XMK'                 oMM.
-.MMc             ''^*~l. xMN:                    KM0
-'MM.                   .NMO                      oMM
-.MM,                 .kMMl                       xMN
- KM0               .kMM0' .dl>~,.               .WMd
- 'XM0.           ,OMMK'    OMMM7'              .XMK
-   *WMO:.    .;xNMMk'       NNNMKl.          .xWMx
-     ^ONMMNXMMMKx;          V  'xNMWKkxllox0NMWk'
-         '''''                    ':dOOXXKOxl'
+${c2}         JJJJJJJJ
+      JJJJJJJJJJJJJJ
+    JJJJJJ   =JJJJJJJ
+   JJJJ      =JJJ JJJJ
+   JJJ       =JJJ   JJJ
+  JJJJ       =JJJ   JJJ
+  JJJJJJJJJJJJJJJ   JJJJ
+   JJJJJJJJJJJJJJ   JJJJ
+   JJJJ             JJJJ
+    JJJJJ=          JJJJ
+      JJJJJJJJJJJJJJJJJJJJJJJJJJJJJ=
+        =JJJJJJJJJJJJJJJJJJJJJJJJJJJJJ
+                    JJJJ         =JJJJJJ
+                    JJJJ            =JJJJ
+                    JJJJ   JJJJJJJJJJJJJJ
+                    JJJJ   JJJJJJJJJJJJJJJ
+                    JJJJ   JJJJ       JJJJ
+                     JJJ   JJJJ       JJJ
+                     JJJJJ JJJJ      JJJJ
+                      =JJJJJJJJ   JJJJJJ
+                        JJJJJJJJJJJJJJ
+                           JJJJJJJ=
 EOF
         ;;
 
